gems/sorbet-runtime/lib/types/private/class_utils.rb

```diff
@@ -69,6 +69,27 @@ def to_s
     end
   end
 
+  def self.def_with_visibility(mod, name, visibility, method=nil, &block)
+    mod.module_exec do
+      # Start a visibility (public/protected/private) region, so that
+      # all of the method redefinitions happen with the right visibility
+      # from the beginning. This ensures that any other code that is
+      # triggered by `method_added`, sees the redefined method with the
+      # right visibility.
+      send(visibility)
+
+      if method
+        define_method(name, method)
+      else
+        define_method(name, &block)
+      end
+
+      if block && block.arity < 0 && respond_to?(:ruby2_keywords, true)
+        ruby2_keywords(name)
+      end
```

Comment:
I encapsulated this `ruby2_keywords` call in this method as well, but I am not sure if that's the best thing to do. The `define_method` calls originating from `call_validation` were never changing the `ruby2_keywords` status of methods, but I am not sure if that was because it was forgotten or if it was intentional.

I'd be ok with moving this part out back to the original call sites if needed.

Reply:
Honestly I think you'd be better at telling me where this `ruby2_keywords` call has to go, as Stripe's codebase is still using Ruby 2.7, and so is unlikely to turn up any problems from incorrectly placement of this.